File: core/src/com/symbol/ecs/system/RemoveSystem.java
package com.symbol.ecs.system;

import com.badlogic.ashley.core.Entity;
import com.badlogic.ashley.core.Family;
import com.badlogic.ashley.systems.IteratingSystem;
import com.symbol.ecs.Mapper;
import com.symbol.ecs.component.RemoveComponent;

public class RemoveSystem extends IteratingSystem {

    public RemoveSystem() {
        super(Family.all(RemoveComponent.class).get());
    }

    public void processEntity(Entity entity, float dt) {
        RemoveComponent rem = Mapper.REMOVE_MAPPER.get(entity);

        if (rem.shouldRemove) {
            getEngine().removeEntity(entity);
        }
    }

}

File: core/src/com/symbol/ecs/system/RemoveSystem.kt
package com.symbol.ecs.system

import com.badlogic.ashley.core.Entity
import com.badlogic.ashley.core.Family
import com.badlogic.ashley.systems.IteratingSystem
import com.symbol.ecs.Mapper
import com.symbol.ecs.component.RemoveComponent

class RemoveSystem : IteratingSystem(Family.all(RemoveComponent::class.java).get()) {

    override fun processEntity(entity: Entity?, deltaTime: Float) {
        val rem = Mapper.REMOVE_MAPPER.get(entity)

        if (rem.shouldRemove) {
            engine.removeEntity(entity)
        }
    }

}